corehq/apps/hqcase/api/get_list.py

```diff
@@ -76,16 +78,25 @@ def get_list(domain, couch_user, params):
         params_string = b64decode(params['cursor']).decode('utf-8')
         params = QueryDict(params_string, mutable=True)
         # QueryDict.pop() returns a list
+        include_deprecated = params.pop(INCLUDE_DEPRECATED, ['False'])[0].lower().capitalize() == 'True'
         last_date = params.pop(INDEXED_AFTER, [None])[0]
         last_id = params.pop(LAST_CASE_ID, [None])[0]
         query = _get_cursor_query(domain, params, last_date, last_id)
     else:
         params = params.copy()  # Makes params mutable for pagination below
+        include_deprecated = params.pop(INCLUDE_DEPRECATED, ['False'])[0].lower().capitalize() == 'True'
         query = _get_query(domain, params)
 
     if not couch_user.has_permission(domain, 'access_all_locations'):
         query = query_location_restricted_cases(query, domain, couch_user)
 
+    # Cases with deprecated case types should not be returned if the flag is not specified
+    if not include_deprecated:
+        deprecated_case_types = get_data_dict_deprecated_case_types(domain)
+        query = query.filter(
+            filters.NOT(filters.term('type.exact', deprecated_case_types))
+        )
```

Comment:
Can you implement this instead as a `SIMPLE_FILTER`?  That's where other URL parameters are turned into ES filters.  There's already a `_to_boolean` component used there that should be re-used for consistent behavior between boolean filters.

Reply:
The problem I had when initially trying to implement this as a `SIMPLE_FILTER` was that when all the filters get added in `_get_filter()`, only the query param value is passed in as a function input. Being able to determine whether to include deprecated case type cases would require the domain as well. To accommodate this, `_get_filter()` would have to be made less generic as follows:
```
...
elif key in SIMPLE_FILTERS:
    if key == 'include_deprecated':
        return SIMPLE_FILTERS[key](domain, val)    
    return SIMPLE_FILTERS[key](val)
...
```
@esoergel Do you think this would be fine? Or did you perhaps have a different implementation in mind?

Reply:
oh yeah that does make things harder, that might require restructuring the filters maps into a stateful object to do properly.  That "less generic" solution you shared seems like a reasonable compromise to me, though.

Reply:
Thanks for the feedback, I have addressed it in [this](https://github.com/dimagi/commcare-hq/pull/33089/commits/f23f4eb62edd725b6867a7153fa5ea366db5f50b) commit. This change should also address the point raised in [this](https://github.com/dimagi/commcare-hq/pull/33089#discussion_r1251594762) comment regarding defaulting to including cases with deprecated case types.